nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,503 @@
+          authentication = "anonymous"
+          c2s_require_encryption = false
+          admins = { "focus@auth.${cfg.hostName}" }
+        '';
+      };
+      virtualHosts."auth.${cfg.hostName}" = {
+        enabled = true;
+        domain = "auth.${cfg.hostName}";
+        extraConfig = ''
+          authentication = "internal_plain"
+        '';
+      };
+    };
+    systemd.services.prosody.serviceConfig = mkIf cfg.prosody.enable {
+      EnvironmentFile = [ "/var/lib/jitsi-meet/secrets" ];
+      SupplementaryGroups = [ "jitsi-meet" ];
+    };
+
+    users.groups.jitsi-meet = {};
+    systemd.tmpfiles.rules = [
+      "d '/var/lib/jitsi-meet' 0750 root jitsi-meet - -"
+    ];
+
+    systemd.services.jitsi-meet-init-secrets = {
+      wantedBy = [ "multi-user.target" ];
+      before = [ "jicofo.service" "jitsi-videobridge2.service" ] ++ (optional cfg.prosody.enable "prosody.service");
+      serviceConfig = {
+        Type = "oneshot";
+      };
+
+      script = let
+        secrets = [ "VIDEOBRIDGE_SECRET" "JICOFO_COMPONENT_SECRET" "JICOFO_USER_SECRET" ];
+      in
+      ''
+        if [ ! -f /var/lib/jitsi-meet/secrets ]; then
+          ${concatMapStringsSep "\n" (var: ''
+            echo "${var}=$(tr -dc a-zA-Z0-9 </dev/urandom | head -c 64)" >> /var/lib/jitsi-meet/secrets
+          '') secrets}
+
+          chown root:jitsi-meet /var/lib/jitsi-meet/secrets
+          chmod 640 /var/lib/jitsi-meet/secrets
+        fi
+
+      ''
+      + optionalString cfg.prosody.enable ''
+        source /var/lib/jitsi-meet/secrets
+        ${config.services.prosody.package}/bin/prosodyctl register focus auth.${cfg.hostName} "$JICOFO_USER_SECRET"
+        ${config.services.prosody.package}/bin/prosodyctl register jvb auth.${cfg.hostName} "$VIDEOBRIDGE_SECRET"
+
+        cd ${config.services.prosody.dataDir}
+
+        # regenerate the certificates after 1 month - they are valid for a year, that leaves 11 months for service restart
+        find ${cfg.hostName}.crt -mtime +30 -delete || true
+
+        # generate self-signed certificates
+        if [ ! -f ${cfg.hostName}.crt ]; then
+          export PATH=$PATH:${pkgs.openssl.bin}/bin
+          rm -f {,auth.}${cfg.hostName}.{crt,key,cnf}
+          # not sure this one is even used but it avoids some errors in logs
+          echo | ${config.services.prosody.package}/bin/prosodyctl cert generate ${cfg.hostName}
+          # used for jicofo<->prosody user connection
+          echo | ${config.services.prosody.package}/bin/prosodyctl cert generate auth.${cfg.hostName}
+        fi
+
+        # don't know how to add it to the system-wide ceritificate bundle, instead convert it to java truststore and pass it to jicofo
+        rm -f /var/lib/jitsi-meet/prosody-ca.jks
+        ${pkgs.jre}/bin/keytool -importcert -file auth.${cfg.hostName}.crt -keystore /var/lib/jitsi-meet/prosody-ca.jks -storepass javapls -noprompt
+        chown root:jitsi-meet /var/lib/jitsi-meet/prosody-ca.jks
```

Comment:
If using ACME certificates, we also wouldn't need to fiddle with the Java Keystore.